core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -104,27 +104,30 @@ object ZkUtils {
    * Get calls that only depend on static paths
    */
   def getTopicPath(topic: String): String = {
-    ZkUtils.BrokerTopicsPath + "/" + topic
+    getZkPath(ZkUtils.BrokerTopicsPath, topic)
   }
 
   def getTopicPartitionsPath(topic: String): String = {
-    getTopicPath(topic) + "/partitions"
+    getZkPath(getTopicPath(topic), "partitions")
   }
 
   def getTopicPartitionPath(topic: String, partitionId: Int): String =
-    getTopicPartitionsPath(topic) + "/" + partitionId
+    getZkPath(getTopicPartitionsPath(topic), partitionId)
 
   def getTopicPartitionLeaderAndIsrPath(topic: String, partitionId: Int): String =
-    getTopicPartitionPath(topic, partitionId) + "/" + "state"
+    getZkPath(getTopicPartitionPath(topic, partitionId), "state")
     
   def getEntityConfigRootPath(entityType: String): String =
-    ZkUtils.EntityConfigPath + "/" + entityType
+    getZkPath(ZkUtils.EntityConfigPath, entityType)
 
   def getEntityConfigPath(entityType: String, entity: String): String =
-    getEntityConfigRootPath(entityType) + "/" + entity
+    getZkPath(getEntityConfigRootPath(entityType), entity)
 
   def getDeleteTopicPath(topic: String): String =
-    DeleteTopicsPath + "/" + topic
+    getZkPath(DeleteTopicsPath, topic)
+
+  def getZkPath(nodes: Any*): String =
```

Comment:
Thank you @SinghAsDev for your feedback. Since this it somewhat tied into your general concern about the change (as [commented](https://github.com/apache/kafka/pull/1025#issuecomment-193481685) on the PR) let's wait for clarification from @granthenke.